src/Symfony/Component/Cache/Traits/PdoTrait.php

```diff
@@ -0,0 +1,409 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Traits;
+
+use Symfony\Component\Cache\Adapter\DoctrineDbalAdapter;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+use Symfony\Component\Cache\Marshaller\DefaultMarshaller;
+use Symfony\Component\Cache\Marshaller\MarshallerInterface;
+
+/**
+ * @internal
+ */
+trait PdoTrait
+{
+    private const MAX_KEY_LENGTH = 255;
+    protected const NS_SEPARATOR = ':';
+
+    private MarshallerInterface $marshaller;
+    private \PDO $conn;
+    private string $dsn;
+    private string $driver;
+    private string $serverVersion;
+    private string $table = 'cache_items';
+    private string $tagsTable = 'cache_tags';
+    private string $idCol = 'item_id';
+    private string $dataCol = 'item_data';
+    private string $lifetimeCol = 'item_lifetime';
+    private string $timeCol = 'item_time';
+    private string $tagCol = 'item_tag';
```

Comment:
This should not be in the trait but in the tag-aware adapter IMO.

Reply:
Right. Fixed.